functions.py

```diff
@@ -123,14 +123,22 @@ def acos(x):
     if x > 1 or x < -1:
         raise ValueError(_('acos(x) only defined for x E [-1,1]'))
     else:
-        return _inv_scale_angle(math.acos(x))
+        x = _inv_scale_angle(math.acos(x))
+        n = significant(x)
+        if n > 6:
```

Comment:
Rounding the number to 6 significant digits would reduce the accuracy of the result, something that a user would not like if he sets the rounding off value to, say 15

Reply:
@pro-panda Yes, Hence I've removed it in a later patch, after further discussion.